### DISCUSSION TITLE
Adding Descriptions on each of the choices or options

### DISCUSSION THREAD

**Discussion by Tobies-Cormac** [DisnakeDev/disnake#981] [Q&A]
Is there any way to add descriptions to to choices? 
Currently i'm testing out a slash command which i'm able to add the description for command and also description for the required variable input. However i can't find a way to add a description to each of the options. I searched through web multiple time and i just can't find the solution. may be i can't read the document correctly. 
So please if u can help.

Here is the code that i'm dealing with :
```
@bot.slash_command(description="Hey")

async def blep(
    inter: discord.ApplicationCommandInteraction,
    animal: str = commands.Param(choices={"Dog": "dog", "Cat": "cat", "Penguin": "penguin"}, description="Hello")
):
    await inter.response.send_message(animal)
```
Which i'm getting now :

![Screenshot (1007)](https://user-images.githubusercontent.com/62227728/223441418-635f85c2-564e-4091-adc6-fd15b3426f1c.png)

I would like to add description like this:
![Screenshot (1008)](https://user-images.githubusercontent.com/62227728/223441597-fd5fb801-3132-40ee-babf-da6bed59a35a.png)

**Comment by Snipy7374** [comment]
Hi, there are multiple ways to add a description to an option; as mentioned [here](https://docs.disnake.dev/en/stable/ext/commands/slash_commands.html#docstrings) in the docs you could use docstrings writed in a certain way.

e.g

```py
@commands.slash_command()
async def test(self, inter, name):
	"""
	A test command
	
	Parameters
	------------------
	name: a name
	"""
	...
```

another way would be to use [`disnake.ext.commands.ParamInfo`](https://docs.disnake.dev/en/stable/ext/commands/api.html#disnake.ext.commands.ParamInfo)

e.g 
```py
@commands.slash_command()
async def test(self, inter, name: str = commands.ParamInfo(name="name", description="a user name")):
	...
```

edit: sorry if the indentation looks weird but I'm on mobile

**Reply by Sharp-Eyes** [reply]
I think they're specifically referring to descriptions for individual options choices; i.e. `Dog`/`Cat`/`Penguin` in their example.

However, I'm fairly certain that those don't support descriptions. Going by [the dAPI docs' entry on option choices](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-choice-structure), they only support name, name localisations, and value.

**Reply by Snipy7374** [reply]
> I think they're specifically referring to descriptions for individual options choices; i.e. `Dog`/`Cat`/`Penguin` in their example.
> 
> However, I'm fairly certain that those don't support descriptions. Going by [the dAPI docs' entry on option choices](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-choice-structure), they only support name, name localisations, and value. 

ahhh, I misunderstood it 😔

**Reply by Tobies-Cormac** [reply]
> I think they're specifically referring to descriptions for individual options choices; i.e. `Dog`/`Cat`/`Penguin` in their example.
> 
> However, I'm fairly certain that those don't support descriptions. Going by [the dAPI docs' entry on option choices](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-choice-structure), they only support name, name localisations, and value.

So there is no way to add description for each choices at the moment?

**Reply by Sharp-Eyes** [reply]
Correct. The discord API does not support this.
